File: src/pizza_split_legacy.c
#include <stdio.h>
#include <stdlib.h>
#include <limits.h>
#include <errno.h>

#define MAX_PIZZAS 1000
#define MAX_PEOPLE 1000

int main(int argc, char *argv[]) {
    long num_pizzas_long;
    long num_people_long;
    int num_pizzas;
    int num_people;
    double pizza_price = 10.0; // 10€ per pizza
    double total_bill;
    double cost_per_person;
    char *endptr;
    
    printf("=== Pizza Delivery Bill Splitter ===\n\n");
    
    // Check if correct number of arguments provided
    if (argc != 3) {
        printf("Usage: %s <number_of_pizzas> <number_of_people>\n", argv[0]);
        printf("Example: %s 3 4\n", argv[0]);
        printf("Note: Each pizza costs %.0f€\n", pizza_price);
        return 1;
    }
    
    // Parse command line arguments with proper error checking
    errno = 0;
    num_pizzas_long = strtol(argv[1], &endptr, 10);
    if (errno != 0 || *endptr != '\0' || num_pizzas_long <= 0 || num_pizzas_long > MAX_PIZZAS) {
        printf("Error: Number of pizzas must be a positive integer (1-%d).\n", MAX_PIZZAS);
        return 1;
    }
    num_pizzas = (int)num_pizzas_long;
    
    errno = 0;
    num_people_long = strtol(argv[2], &endptr, 10);
    if (errno != 0 || *endptr != '\0' || num_people_long <= 0 || num_people_long > MAX_PEOPLE) {
        printf("Error: Number of people must be a positive integer (1-%d).\n", MAX_PEOPLE);
        return 1;
    }
    num_people = (int)num_people_long;
    
    // Calculate the total bill and cost per person
    total_bill = num_pizzas * pizza_price;
    cost_per_person = total_bill / num_people;
    
    // Display the results
    printf("=== Bill Split Results ===\n");
    printf("Number of pizzas: %d\n", num_pizzas);
    printf("Price per pizza: %.0f€\n", pizza_price);
    printf("Total bill: %.2f€\n", total_bill);
    printf("Number of people: %d\n", num_people);
    printf("Cost per person: %.2f€\n", cost_per_person);
    
    return 0;
}

File: python-demo/pizza_split_legacy.py
#!/usr/bin/env python3
"""
Pizza Bill Splitter - Legacy Command Line Version
Simple command-line tool for calculating pizza bill splits
"""

import sys
import argparse
from decimal import Decimal

# Security constants
MAX_PIZZAS = 1000
MAX_PEOPLE = 1000
PIZZA_PRICE = Decimal('10.00')

def validate_positive_int(value: str, name: str, max_val: int) -> int:
    """Validate and convert string to positive integer"""
    try:
        num = int(value)
    except ValueError:
        raise ValueError(f"{name} must be a valid integer")
    
    if num <= 0:
        raise ValueError(f"{name} must be positive")
    
    if num > max_val:
        raise ValueError(f"{name} too large (max {max_val})")
    
    return num

def main():
    """Main entry point for legacy pizza splitter"""
    parser = argparse.ArgumentParser(
        description="Calculate pizza delivery bill split",
        formatter_class=argparse.RawDescriptionHelpFormatter,
        epilog=f"""
Examples:
  {sys.argv[0]} 3 4    # 3 pizzas split among 4 people
  
Note: Each pizza costs €{PIZZA_PRICE}
        """
    )
    
    parser.add_argument(
        'pizzas',
        type=str,
        help=f'Number of pizzas (1-{MAX_PIZZAS})'
    )
    
    parser.add_argument(
        'people',
        type=str,
        help=f'Number of people (1-{MAX_PEOPLE})'
    )
    
    try:
        args = parser.parse_args()
        
        # Validate arguments
        num_pizzas = validate_positive_int(args.pizzas, "Number of pizzas", MAX_PIZZAS)
        num_people = validate_positive_int(args.people, "Number of people", MAX_PEOPLE)
        
        # Calculate totals
        total_bill = num_pizzas * PIZZA_PRICE
        cost_per_person = (total_bill / num_people).quantize(Decimal('0.01'))
        
        # Display results
        print("=== Pizza Delivery Bill Splitter ===\\n")
        print("=== Bill Split Results ===")
        print(f"Number of pizzas: {num_pizzas}")
        print(f"Price per pizza: €{PIZZA_PRICE}")
        print(f"Total bill: €{total_bill}")
        print(f"Number of people: {num_people}")
        print(f"Cost per person: €{cost_per_person}")
        
    except ValueError as e:
        print(f"Error: {e}", file=sys.stderr)
        sys.exit(1)
    except KeyboardInterrupt:
        print("\\nOperation cancelled.", file=sys.stderr)
        sys.exit(1)
    except Exception as e:
        print(f"Unexpected error: {e}", file=sys.stderr)
        sys.exit(1)

if __name__ == "__main__":
    main()
